sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -128,6 +130,46 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
         ShimLoader.setSparkShimProviderClass(conf.shimsProviderOverride.get)
       }
 
+      // Compare if the cudf version mentioned in the classpath is equal to the version which plugin
+      // expects. If there is a version mismatch, throw error. This check can be disabled by
+      // setting this config spark.rapids.cudfVersionOverride=true
+      if (!conf.cudfVersionOverride) {
+        val props = new Properties
+        val cudfClassLoader = classOf[ai.rapids.cudf.ColumnVector].getClassLoader
+        val cudfProperties = cudfClassLoader.
+          getResourceAsStream("cudf-java-version-info.properties")
+
+        if (cudfProperties == null) {
+          throw new RuntimeException("Could not find the properties file in the cudf jar." +
+            " Cannot verify if the cudf version is same as expected by plugin")
+        }
+        props.load(cudfProperties)
+        if (props.get("version") == null) {
+          throw new RuntimeException("Property name not found in " +
+            "cudf-java-version-info.properties file")
+        }
+        val cudfVersion = props.get("version").toString
+
+        val pluginClassLoader = classOf[com.nvidia.spark.SQLPlugin].getClassLoader
+        val pluginResource = pluginClassLoader.
+          getResourceAsStream("rapids4spark-version-info.properties")
+        if (pluginResource == null) {
+          throw new RuntimeException("Could not find the properties file in the plugin jar." +
+            " Cannot verify if the cudf version is same as expected by plugin")
+        }
+        props.load(pluginResource)
+        if (props.get("cudf_version") == null) {
+          throw new RuntimeException("Property name not found in " +
+            "rapids4spark-version-info.properties file")
+        }
+        val expectedCudfVersion = props.get("cudf_version").toString
+
+        if (!cudfVersion.equals(expectedCudfVersion)) {
+          throw new IllegalArgumentException("Cudf version in the classpath is different. Found " +
+            cudfVersion + ", Plugin expects " + expectedCudfVersion)
```

Comment:
@jlowe  Do we want to add in something about setting the override?  We have marked that config as internal.  For me I would rather tell them if they ran into an issue that this config exists but only use it if you really know what you are doing, but I am OK with not saying anything too.

Reply:
Hmm.  If we point the user to the internal config when this exception is thrown, we should be explicit that it is an untested and unsupported configuration they are running.

Reply:
I am fine either way.  I just know it is a lot simpler to have the config I have to set right in front of me than to have to go look it up in the code.  I doubt I will need the config often (at least I really hope I will not need it), so I am fine without it.